src/server/game/Entities/Creature/Creature.h

```diff
@@ -579,6 +587,9 @@ class TC_GAME_API Creature : public Unit, public GridObject<Creature>, public Ma
         uint32 _gossipMenuId;
         Optional<uint32> _trainerId;
         float _sparringHealthPct;
+
+        // The Summon API which controls the behavior of summons
+        std::unique_ptr<SummonInfo> _summonInfo;
```

Comment:
is there any case at all where we would be fine at allowing multiple ownership at some point due to scripting reasons for this to be accessible? just asking in case it'd be convenient to switch to shared_ptr instead for the future

Reply:
There is no such thing such as multiple ownership. You call the spell effect, you summon the creature, you are the summoner. If someone were to take the summon away from you, you would charm that creature (which is a whole different mechanic).

As for the optional --> I don't want to bloat the memory of the creature object any further and keep the summon API nicely contained in its own cozy space on the heap. By using a pointer, we just use additional 64 bit for the creature instead of shoving in the whole size of the API